### ISSUE TITLE
Debug Output for build 1.1.0.40

### ISSUE THREAD

**Issue by Ahiga** [Archomeda/csl-ambient-sounds-tuner#11]
Hi Archomeda. I unsubscribed to the old version of ASTuner and subscribed to the new version. I restarted CS and received the following Debug Output window:

A.
1. I disabled ScrollableToolBar and TerraformTool, restarted CS but recieved the below message. 
2. Unsubscribed to ASTuner, kept others disabled, restarted CS, subscribed ASTuner, restarted CS and received debug output again. Screenshot below
![image 1](https://cloud.githubusercontent.com/assets/12837457/8097254/f0bff952-0fad-11e5-9f45-a1cbb619dbe4.jpg)

B.
I went into CS and ASTuner worked with the above disabled. The volume controls appear to work as intended. When entering from or exiting to the CSe main menu I received the Debug Output window with a huge list. I realize this may not be of any value to you but I wanted to illustrate what I was getting.
Screenshot below:
![image 002](https://cloud.githubusercontent.com/assets/12837457/8097321/e23d23e0-0fae-11e5-99c8-0e741c34f890.png)
![image 003](https://cloud.githubusercontent.com/assets/12837457/8097322/e67762ea-0fae-11e5-98f0-d4c881831232.png)
![image 004](https://cloud.githubusercontent.com/assets/12837457/8097325/ed214868-0fae-11e5-87a3-8c20cd5ff1d9.png)
![image 005](https://cloud.githubusercontent.com/assets/12837457/8097328/f1b6fea4-0fae-11e5-94c5-7ec93c51a73e.png)

Upon exiting to the main menu.
![image 7](https://cloud.githubusercontent.com/assets/12837457/8097615/8dcba5da-0fb2-11e5-872d-3f8a2a3b9997.jpg)

BTW: Thanks for your explaination, that explained a lot. Sorry for my delayed response, I'm UTC+5 hrs and couldn't get to this until this evening.

System Spec:
Intel Core i7-3930K Sandy Bridge-E 3.2GHz (3.8GHz Turbo) LGA 2011 Six-Core
MSI R9 290X LIGHTNING 4GB 512-Bit GDDR5 PCI Express 3.0
ASRock X79 Extreme6 LGA 2011 Intel X79 SATA 6Gb/s USB 3.0
COOLER MASTER HAF X Black Steel/ Plastic ATX w/tons of cooling
CORSAIR Hydro Series H100
Dual ASUS Black 24" 2ms HDMI LED monitors
G.SKILL Ripjaws Z Series 16GB (4 x 4GB)
Dual Corsair Force GT Series 180GB SATA III Internal SSD's 
CORSAIR HX Series 750W SLI Ready CrossFire Ready 80 PLUS GOLD Certified Modular PSU
Logitech G510s Keyboard
Logitech G13 gamepad
Logitech G700s Laser Mouse

**Comment by Ahiga** [comment]
Update: I disabled ARIS mods and enabled the ScrollableToolbar and the TerraformTool. Then I re-enabled the ARIS mods. While I still received the above Debug Output the above mods worked just fine.

**Comment by Archomeda** [comment]
Hey! Thanks for the detailed report.

After posting my comment on the workshop, I decided to look again at my code that checks for known incompatible mods. And your screenshots confirm what I think that happens.

The only mod that I've listed as incompatible, is [SilenceObnoxiousSirens](http://steamcommunity.com/sharedfiles/filedetails/?id=421527612). I think you are subscribed to that mod, and therefore it's installed, but you don't have it enabled.
I've made a small mistake somewhere. My mod first gets the list of installed mods that are incompatible, and then filters it to contain only the installed mods that are enabled as well. But it's determined before filtering if this warning should be shown, but it contains the list after filtering, if that makes any sense. Hence why if you have that mod installed you get that warning, but it contains no list of mods since it's filtered.

This is all under the assumption you've got that mod installed :smile:. If it's incorrect, I'd be glad to hear it, since that means the problem lies somewhere else. If it is correct, let me know so I can work on a fix.

And again, if there are no mods listed within that warning, you can just ignore it, it's a false positive. Although I agree that it's quite confusing.

**Comment by Ahiga** [comment]
You're very welcome, Arch, this is the least I could do considering the enormous amount of work you've put into this mod.
You are correct, I do (did) have SilenceObnoxiousSirens installed and subscribed but it was disabled. I have removed it completely (about :30 min ago) and have not received any Debug Output reports. All is performing perfectly now. I ahve also re-enabled the ARIS mods also without a hitch.
I hope this helps.
If you'd like me to test run any updates prior to posting I'd be happy to run the gauntlet. I'm not a coder but I understand software and rigs by far better than quantum mechanics. LOL

**Comment by Archomeda** [comment]
Haha. Funny how that works right?

I'll work on a fix later today. It shouldn't take too long to fix it, as the code where this happens is pretty minimal. And thanks for the offer to test it beforehand, I'll let you know when it's ready.

**Comment by Archomeda** [comment]
As you probably already have seen, I've created a separate PR. Once AppVeyor is done building, you can find the new files under [the artifacts of build 1.1.0.43](https://ci.appveyor.com/project/Archomeda/csl-ambient-sounds-tuner/build/1.1.0.43/artifacts). You can download the first zip, extract it to the mods folder (should be `Steam/steamapps/workshop/content/255710/455958878`) and replace the files. Back up the existing files just in case.

Let me know if this solves the issue for you.

**Comment by Ahiga** [comment]
Hello Archomeda:

I just ran 1.1.0.43 without any problems at all. No Debug output, system.nullReferences, or filenames unknown. I ran CS for about 30 minutes and everything was solid. All of the sliders appeared to work as intended.
I will run it longer after I've settled in and grab some dinner. If anything pops up, which I doubt will at this point, I'll let you know.

BTW: I incorrectly said I was UTC +5... duuuh I'm UTC -4 (Connecticut, USA). I think I flatlined when typing.

**Comment by Archomeda** [comment]
I waited for a little bit longer, but I guess you don't run into any issues anymore, which is good! I'll merge it and push a new version after I've hotfixed something.

Also, that's 9 hours difference :p

**Comment by Ahiga** [comment]
Hello Archomeda: I updated AS to your new Build 1.1.0.46 and everything is still solid and running great. It appears others are reporting the same on Steams Workshop. Thanks so much for your rapid responses.
Wow 9 hrs? I think that is so cool meeting people from all over the world. It's currently 7:41 local (11:41 UTC).
I hope you have a great weekend.

**Comment by Archomeda** [comment]
Good :smile: and you're welcome!

And I was referring to your UTC+5 vs UTC-4 with that 9 hour difference. My timezone is currently CEST, which is UTC+2. It's still the other side of the ocean though.
